src/coreclr/jit/helperexpansion.cpp

```diff
@@ -442,6 +442,8 @@ PhaseStatus Compiler::fgExpandThreadLocalAccess()
         return result;
     }
 
+    // Always expand for NativeAOT
+    const bool isNativeAOT = IsTargetAbi(CORINFO_NATIVEAOT_ABI);
     if (opts.OptimizationDisabled())
```

Comment:
If `OptimizationDisabled` can be forced in a scenario where ILC produces inlined TLS, then we should not remove the support for Windows/x64 in `EmitInlineTLSAccess` as that code would still be reachable.

Reply:
Yep, just wrote about it in https://github.com/dotnet/runtime/pull/89472#discussion_r1449611726

Reply:
i will revert this change.